upgrades/steps_27.go

```diff
@@ -79,14 +89,97 @@ func stepsFor27() []Step {
 	}
 }
 
+func setLogOwnerCorrectLogPermissions(filename string) error {
+	group, err := user.LookupGroup("adm")
+	if err != nil {
+		return err
+	}
+	gid, err := strconv.Atoi(group.Gid)
+	if err != nil {
+		return err
+	}
+	usr, err := user.Lookup("syslog")
+	if err != nil {
+		return err
+	}
+	uid, err := strconv.Atoi(usr.Uid)
+	if err != nil {
+		return err
+	}
+	if err := os.Chmod(filename, 0640); err != nil {
+		return err
+	}
+	return os.Chown(filename, uid, gid)
+}
+
+func setFolderPermissionsToAdm(dir string) error {
+	err := filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if err := setLogOwnerCorrectLogPermissions(info.Name()); err != nil {
+			return err
+		}
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	logger.Infof("Successfully changed permissions of /var/log/juju")
+	return nil
+}
+
 // This adds upgrade steps, we just rewrite the default values which are set before.
 // With this we can make sure that things are changed in one default place
 func resetLogPermissions(context Context) error {
-	sysdManager := service.NewServiceManagerWithDefaults()
-	err := sysdManager.WriteServiceFiles()
+	if context.AgentConfig().Tag().Kind() == names.UnitTagKind {
```

Comment:
Probably easier to just test for MachineTagKind, and log the skip:

```go
tag := context.AgentConfig().Tag()
if tag.Kind() == names.MachineTagKind {
    logger.Infof("skipping agent %q, not a machine", tag.String())
￼    return nil
}
```

Reply:
:+1: